### PR TITLE
Adding PHPStan to CI build

### PR DESCRIPTION
**Pull request by moufmouf**
### Context

PHPStan is a static analysis tool. It can be used to detect issues automatically in PHP code.

### What has been done

I added PHPStan in the Travis builds.

The first commit of this PR enables "level 0". I haven't modified the code so the build is expected to fail.

In the following commits, I plan to fix "level 0" bugs, then increase to level 1, fix level 1 bugs, etc... up to level 7 (the maximum level in PHPStan).

### CODE REVIEW

**Comment by moufmouf** [comment]
Level 0 issues found in code: 

```
------ ------------------------------------------------------------------------ 
  Line   VCR/CodeTransform/AbstractCodeTransform.php                             
 ------ ------------------------------------------------------------------------ 
  10     Class php_user_filter referenced with incorrect case: PHP_User_Filter.  
 ------ ------------------------------------------------------------------------ 
 ------ ------------------------------------------------------------------------- 
  Line   VCR/Request.php                                                          
 ------ ------------------------------------------------------------------------- 
  61     Parameter $requestMatchers of method VCR\Request::matches() has invalid  
         typehint type callable.                                                  
 ------ ------------------------------------------------------------------------- 
 ------ ------------------------------------------------------------------- 
  Line   VCR/Util/CurlHelper.php                                            
 ------ ------------------------------------------------------------------- 
  16     Array has 2 duplicate keys with value 2097166 (CURLINFO_FILETIME,  
         CURLINFO_FILETIME).                                                
 ------ ------------------------------------------------------------------- 
 ------ --------------------------------------------------------------- 
  Line   VCR/Util/SoapClient.php                                        
 ------ --------------------------------------------------------------- 
  33     Call to an undefined static method SoapClient::__construct().  
 ------ --------------------------------------------------------------- 
```

See: https://travis-ci.org/php-vcr/php-vcr/jobs/444770027

The next commit will fix level 0 issues.

File: phpstan.neon [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#parameters:
-#    ignoreErrors:
+parameters:
+    ignoreErrors:
+        - "#Call to an undefined static method SoapClient::__construct\\(\\)#"
```
Comment (renatomefi) [inline phpstan.neon]: That's weird, it was a constructor right? http://us2.php.net/manual/en/soapclient.construct.php
Or should we refactor to call the SoapClient method?

Comment (moufmouf) [reply phpstan.neon]: Yes, it has a constructor. This is a PHPStan bug.
That is why I added the PHPStan in the `ignoreErrors` section.

I should probably open an issue in the PHPStan repo regarding this.

Comment (renatomefi) [reply phpstan.neon]: Maybe someone created an extension for it

Comment (moufmouf) [reply phpstan.neon]: Ho, no, I just discovered it.

It is most definitely a bug in PHPStan. The soap extension is enabled and the error is still present.

I think Ondrej nailed the problem here: https://github.com/phpstan/phpstan/issues/618#issuecomment-347240188

I'm pretty sure the issue will be solved in the next PHPStan release. We can keep the `ignoreErrors` parameter and remove it when the bug is fixed in PHPStan.

**Comment by JeroenVanOort** [comment]
I've updated your PR to the latest PHPStan version. I plan to merge it as it is now. I do wish to raise the PHPStan level, but I think it would be better to do that in separate PR's.

**Comment by moufmouf** [comment]
Excellent! Thanks for taking this over!

Actually, I raised PHPStan level to 7 in this PR (https://github.com/php-vcr/php-vcr/pull/259) which was supposed to really be a new major version (or at least a big minor version) but never got merged. Check it out, I spent a good week working on it, #259 contains a number of fixes done by others that I merged to have something usable + PHP 7 type-hints...

**Comment by JeroenVanOort** [comment]
I will definitely check it out; it was already on my to do list and hope to get it merged soon.

**Pull request merged**